lib/puppet/catalog-diff/compilecatalog.rb

```diff
@@ -97,14 +97,14 @@ def compile_catalog(node_name, server, certless)
       end
 
       uri = URI("https://#{server}:#{port}#{endpoint}")
-      Puppet.debug("Connecting to server: #{uri}")
```

Comment:
I don't think that this debug statement is required. The http client itself will log this at debug level already.